modules/core/include/opencv2/core/hal/intrin_wasm.hpp

```diff
@@ -3337,25 +3337,19 @@ inline v_uint16x8 v_mul_hi(const v_uint16x8& a, const v_uint16x8& b)
 
 inline v_int32x4 v_dotprod(const v_int16x8& a, const v_int16x8& b)
 {
-    v_int32x4 a0, a1, b0, b1;
-    v_expand(a, a0, a1);
-    v_expand(b, b0, b1);
-    v128_t c = wasm_i32x4_mul(a0.val, b0.val);
-    v128_t d = wasm_i32x4_mul(a1.val, b1.val);
-    v128_t e = wasm_v8x16_shuffle(c, d, 0,1,2,3,8,9,10,11,16,17,18,19,24,25,26,27);
-    v128_t f = wasm_v8x16_shuffle(c, d, 4,5,6,7,12,13,14,15,20,21,22,23,28,29,30,31);
+    v128_t c = wasm_i16x8_mul(a.val, b.val);
```

Comment:
Wouldn't that reduce the precision of dot product?

Reply:
It will. So @Wenzhao-Xiang fix that in https://github.com/huningxin/opencv/pull/289/files. Now it uses `wasm_i32x4_mul`